crypto/merkle/proof.go

```diff
@@ -43,18 +43,19 @@ func (poz ProofOperators) Verify(root []byte, keypath string, args [][]byte) (er
 	for i, op := range poz {
 		key := op.GetKey()
 		if len(key) != 0 {
-			if !bytes.Equal(keys[0], key) {
-				return cmn.NewError("Key mismatch on operation #%d: expected %+v but %+v", i, []byte(keys[0]), []byte(key))
+			lastKey := keys[len(keys)-1]
```

Comment:
should we check that `len(keys) > 0` especially when we're modifying them later `keys = keys[:len(keys)-1]`

Reply:
KeyPathToKeys ensures that the keys aren't empty, unless there's an error